pkg/agent/cniserver/pod_monitor.go

```diff
@@ -0,0 +1,246 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cniserver
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"time"
+
+	"antrea.io/libOpenflow/openflow15"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+	agenttypes "antrea.io/antrea/pkg/agent/types"
+	"antrea.io/antrea/pkg/util/channel"
+)
+
+const (
+	podNotReadyTimeInSeconds = 30 * time.Second
+	// How long to wait before retrying the processing of an Egress change.
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+)
+
+var (
+	// retryInterval is the interval to re-install Pod OpenFlow entries if any error happened.
+	// Note, using a variable rather than constant for retryInterval because we may use a shorter time in the
+	// test code.
+	retryInterval = time.Second * 5
+)
+
+type unreadyPodInfo struct {
+	podName        string
+	podNamespace   string
+	flowsInstalled bool
+	createTime     time.Time
+}
+
+type podIfaceMonitor struct {
+	podLister         corelisters.PodLister
+	kubeClient        clientset.Interface
+	eventBroadcaster  record.EventBroadcaster
+	record            record.EventRecorder
+	ifaceStore        interfacestore.InterfaceStore
+	ofClient          openflow.Client
+	podUpdateNotifier channel.Notifier
+
+	// unreadyInterfaces is a map to store the OVS ports which is waiting for the PortStatus from OpenFlow switch.
+	// The key in the map is the OVS port name, and its value is unreadyPodInfo.
+	// It is used only on Windows now.
+	unreadyInterfaces map[string]*unreadyPodInfo
+	// unreadyMapMutex is a mutex to guard map unreadyInterfaces.
+	unreadyMapMutex sync.Mutex
+	statusCh        chan *openflow15.PortStatus
+
+	queue workqueue.TypedRateLimitingInterface[string]
+	// statusMapMtx is a mutex to guard ovsPortStatusMap
+	statusMapMtx     sync.Mutex
+	ovsPortStatusMap map[string]uint32
+}
+
+func (m *podIfaceMonitor) Run(stopCh <-chan struct{}) {
+	klog.Info("Starting the monitor to wait for new OpenFlow ports")
+	m.eventBroadcaster.StartStructuredLogging(0)
+	m.eventBroadcaster.StartRecordingToSink(&v1.EventSinkImpl{
+		Interface: m.kubeClient.CoreV1().Events(""),
+	})
+	defer m.eventBroadcaster.Shutdown()
+
+	go wait.Until(m.worker, time.Second, stopCh)
+
+	go wait.Until(func() {
+		ticker := time.Tick(retryInterval)
+		for {
+			select {
+			case status := <-m.statusCh:
+				klog.V(2).InfoS("Received PortStatus message", "message", status)
+				// Update Pod OpenFlow entries only after the OpenFlow port state is live.
+				m.processPortStatusMessage(status)
+			case <-ticker:
+				m.checkUnreadyPods()
```

Comment:
why do we still need the ticker? All we need is to call `queue.AddAfter(podNotReadyTimeInSeconds)` in `addUnreadyPodInterface` if I understand correctly